eodatasets3/prepare/sentinel_sinergise_L1_prepare.py

```diff
@@ -0,0 +1,224 @@
+from pathlib import Path
+from xml.dom import minidom
+import json
+from os import listdir
+from os.path import isfile, join
+import uuid
+import click
+from typing import Dict, Tuple
+from eodatasets3 import DatasetAssembler
+
+SENTINEL_MSI_BAND_ALIASES = {
+    "01": "coastal_aerosol",
+    "02": "blue",
+    "03": "green",
+    "04": "red",
+    "05": "red_edge_1",
+    "06": "red_edge_2",
+    "07": "red_edge_3",
+    "08": "nir_1",
+    "8A": "nir_2",
+    "09": "water_vapour",
+    "10": "swir_1_cirrus",
+    "11": "swir_2",
+    "12": "swir_3",
+}
+
+
+def extract_metadata_from_product_info(product_path: Path) -> Dict:
+    fp = open(product_path)
+    product = json.loads(fp.read())
+
+    synergise_product_name = product["name"]
+    synergise_product_id = product["id"]
+    timestamp = product["tiles"][0]["timestamp"]
+    utm_zone = product["tiles"][0]["utmZone"]
+    latitude_band = product["tiles"][0]["latitudeBand"]
+    grid_square = product["tiles"][0]["gridSquare"]
```

Comment:
Are we sure that there's always only one tile in the list?

Perhaps add a safety check so we don't risk quietly missing data in the future:
```suggestion
    if len(product["tiles"]) > 1:
        raise NotImplementedError("No support for multi-tiled products yet")
    [tile] = product["tiles"]
    timestamp = tile["timestamp"]
    utm_zone = tile["utmZone"]
    latitude_band = tile["latitudeBand"]
    grid_square = tile["gridSquare"]
```

Reply:
added